tests/istanbul/eip152_blake2/test_blake2.py

```diff
@@ -0,0 +1,557 @@
+                t_0="1000000000000000",
+                t_1="0000000000000000",
+            ),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0xbc112be5618b20d24be64c9e1c6efd63fea38cc79d53692fad6568b16e953eb6",
+                data_2="0x128c1ec8ffaf9a2d69e3cb043d6e11e1c7afd48573311052b6e7ec0960371186",
+            ),
+            id="EIP-192-1024-16",
+        ),
+        pytest.param(
+            Blake2Input(
+                rounds=16,
+                h="48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b",
+                m="6162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d900000000000000",
+                t_0="7800000000000000",
+                t_1="0000000000000000",
+            ),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x7df6f69476a03ae29e944814846460b058d1762fffe77f938ea723d1033de0d5",
+                data_2="0xbb1f8234bd73afaf955622fa2cdde95594577a8d53191908eb69b316a53c985b",
+            ),
+            id="EIP-192-0016-120",
+        ),
+        pytest.param(
+            Blake2Input(
+                rounds=32,
+                h="48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b",
+                m="6162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d900000000000000",
+                t_0="7800000000000000",
+                t_1="0000000000000000",
+            ),
+            ExpectedOutput(
+                call_return_code="0x01",
+                data_1="0x38dfb111db6c59c39c9bea26c8c872620d89dec22fd7da93c47d0708a3973f52",
+                data_2="0x2858fd9c60fae53f366d7e2820040a8662b336de6d859764f20747acbb8999fe",
+            ),
+            id="EIP-192-0032-120",
+        ),
+        pytest.param(
+            Blake2Input(
+                rounds=64,
+                h="48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5d182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b",
+                m="6162636465666768696a6b6c6d6e6f707172737475767778797a7b7c7d7e7f808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9fa0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3b4b5b6b7b8b9babbbcbdbebfc0c1c2c3c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d900000000000000",
+                t_0="7800000000000000",
```

Comment:
here the original test seems to be incorrect as it actually passes length byte of 30 not 78
you can debug original call with geth and retesteth
```
rt blake2BFiller.yml  -d test-0064-120 --vmtraceraw out
```
search for line with "CALL"  a line above will have memory and the call parameters are there. 
so far I see that original test is mistaken because it was using complex logic 
```
// The message starts at byte 0x44
           if gt(msgLen, 0) {
               for { } gt(msgLen, 0) { } {
                 // Write a byte
                 mstore8(add(0x44, msgLen), add(0x60, msgLen))
                 msgLen := sub(msgLen, 1)
               }
           }
```

Reply:
if you use unix, retesteth nightly bin available here http://retesteth.ethdevops.io/retesteth

Reply:
Thanks for the pointers, I think you are saying the `t_0` value here should be `78`? Should the expected outputs be changed to the values given in the pytest errors I'm getting?

Reply:
yes. because the vector name says 120 (0x78) but in actual test it passes 0x30. I assume ori mistaken in that test logic. therefor python version is correct

Reply:
Sounds good, just wanted to be sure. Thanks!